zerver/lib/import_realm.py

```diff
@@ -1235,6 +1237,18 @@ def do_import_realm(import_dir: Path, subdomain: str, processes: int = 1) -> Rea
         update_model_ids(RealmUserDefault, data, "realmuserdefault")
         bulk_import_model(data, RealmUserDefault)
 
+    if "zerver_scheduledmessage" in data:
+        fix_datetime_fields(data, "zerver_scheduledmessage")
+        re_map_foreign_keys(data, "zerver_scheduledmessage", "sender", related_table="user_profile")
+        re_map_foreign_keys(data, "zerver_scheduledmessage", "recipient", related_table="recipient")
+        re_map_foreign_keys(
+            data, "zerver_scheduledmessage", "sending_client", related_table="client"
+        )
+        re_map_foreign_keys(data, "zerver_scheduledmessage", "stream", related_table="stream")
+        re_map_foreign_keys(data, "zerver_scheduledmessage", "realm", related_table="realm")
+        update_model_ids(ScheduledMessage, data, "scheduledmessage")
+        bulk_import_model(data, ScheduledMessage)
```

Comment:
I suspect for this to be correct, we probably need to call `fix_message_rendered_content` and `fix_upload_links` (possibly with some tweaks) -- otherwise, the rendered Markdown will reference incorrect user IDs.

Reply:
Yeah, good point. I added `fix_message_rendered_content` logic to the commit, tweaking the function slightly. Tested manually, `data-user-id` in the rendered content gets updated appropriately. 

Attachments are more tricky because we currently can't even export them, see the comment that I added:
```
        # TODO: We should be running fix_upload_links(data, "zerver_scheduledmessage")
        # here after tweaking it to handle the data structure, but currently that's useless
        # since our export logic in fetch_attachment_data is not able to export Attachments
        # for scheduled messages.
```

This is because we export them like this:
```
def fetch_attachment_data(
    response: TableData, realm_id: int, message_ids: Set[int]
) -> List[Attachment]:
    attachments = list(
        Attachment.objects.filter(realm_id=realm_id, messages__in=message_ids).distinct()
    )
   ....
```

But Attachements for ScheduledMessage don't and can't have anything in `.messages`. So we can't even link them to ScheduledMessages we're exporting in any easy way I think? And we can't just export all `Attachments` in the realm with empty .messages because I think that might risk exporting old, orphaned attachments that should be private?

@amanagr Just a ping in case I'm missing something here since I'm not that familiar with `ScheduledMessage` logic